snapd.go

```diff
@@ -249,13 +249,11 @@ func action(ctx *cli.Context) {
 	setMaxProcs(cfg.GoMaxProcs)
 
 	c := control.New(cfg.Control)
-
-	coreModules = []coreModule{}
-
-	coreModules = append(coreModules, c)
 	s := scheduler.New(cfg.Scheduler)
+	c.SetTaskScheduler(s)
 	s.SetMetricManager(c)
-	coreModules = append(coreModules, s)
+
+	coreModules = []coreModule{c, s}
```

Comment:
I'm not sure what the best way is. My concern isn't with where control gets a reference to scheduler, but that it has one at all. It seems to go against the idea of them being separate packages.

pluginControl.CollectMetrics gets a list of metrics to collect, if it's dynamic it could expand that to include all metrics currently in the metric catalog (including plugins loaded since the last task fire).  The example in the linked issue was:

>The scenario which presents that:
Step1:  Load `PluginA`
Step2:  Create/start `Task1` with "intel/\*"
=> Task1 is collecting all metrics from PluginA

>Step3a:  Load `PluginB1`
=> Task1 should immediately start collecting the new metrics from PluginB1
In current implementation (PR #803) the metrics are being collected only from pluginA

>Step3b:  Load `PluginB2` which requires config
=> Task1 should immediately start collecting the new metrics from PluginB2, but Task1 does not contain any config for the pluginB2, so after 10 times not receiving the metrics from this plugin, the Task1 will fail  (@jcooklin approved that the initial task which was valid isn’t now so it seems reasonable that it would fail)
 
In this example, does CollectMetrics not have all the information it needs to collect? If I send control a metric `/intel/*`, and the results of that dynamic query has changed from the previous collect,  it would be transparent to the scheduler.

I feel like maybe I am missing a reason why it wouldn't work this way?